sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -582,12 +582,26 @@ def _resolve_ir_placeholders_v2(
             'Input argument supports only the following types: PipelineParam'
             ', str, int, float. Got: "{}".'.format(argument_value))
 
+  component_spec = copy.deepcopy(component_spec)
+
   if not component_spec.name:
     component_spec.name = _components._default_component_name
 
   # task.name is unique at this point.
   pipeline_task_spec.task_info.name = (dsl_utils.sanitize_task_name(task.name))
 
+  # IR only supports constant arguments for parameter inputs.
```

Comment:
The comment isn't 100% accurate, an input can reference a component input parameter which references PipelineJob. RuntimeConfig.parameters, and the value could be missing at compile time.

Reply:
I guess my wording is ambiguous.
What I meant is: "constant argument" *implies*/*requires* "parameter input".
Not the other way around.